pages/rfc/sso-rfc.rst

```diff
@@ -0,0 +1,230 @@
+
+      {"access_token":"eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhaWQiOiJ1cm46b2lkOjIuMTYuODQwLjEuMTEzODgzLjIuNC42LjE6MDAwMDAwMDAiLCJleHAiOjE1ODE0MTI2NjcsImlhdCI6MTU4MTQxMTc2NywiaXNzIjoidXJuOm9pZDoyLjE2Ljg0MC4xLjExMzg4My4yLjQuNi4xOjAwMDAwMDAxIiwic2lkIjoidXJuOm9pZDoyLjE2Ljg0MC4xLjExMzg4My4yLjQuNi4zOjk5OTk5OTk5MCIsInN1YiI6IiJ9.OhniTJcPS45nhJVqXfxsngG5eYS_0BvqFg-96zaWFO90I_5_N9Eg_k7NmIF5eNZ9Xutl1aqSxlSp80EX07Gmk8uzZO9PEReo0YZxnNQV-Zeq1njCMmfdwusmiczFlwcBi5Bl1xYGmLrxP7NcAoljmDgMgmLH0xaKfP4VVim6snPkPHqBdSzAgSrrc-cgVDLl-9V2obPB1HiVsFMYfbHEIb4MPsnPRnSGavYHTxt34mHbRsS8BvoBy3v6VNYaewLr6yz-_Zstrnr4I_wxtYbSiPJUeVQHcD-a9Ck53BdjspnhVHZ4IFVvuNrpflVaB1A7P3A2xZ7G_a8gF_SHMynYSA","expires_in":0,"token_type":""}
+
+#. The client application redirects the user to the jump url with the session token in the URL
+   First retrieve the jump URL:
+
+   .. code-block:: console
+
+     $ curl --location --request GET 'localhost:1323/api/endpoints?orgIds=urn:oid:2.16.840.1.113883.2.4.6.1:00000001&type=urn:ietf:rfc:3986:urn:oid:1.3.6.1.4.1.54851.1:nuts-sso-jump-endpoint'
+
+     [{
+          "URL": "http://sso.nootenboom.local",
+          "endpointType": "urn:ietf:rfc:3986:urn:oid:1.3.6.1.4.1.54851.1:nuts-sso-jump-endpoint",
+          "identifier": "a5c4afba-1393-48dc-b506-e0b6cc969094",
+          "status": "active",
+          "version": "0.1.0"
+      }]
+
+   And then redirect the user:
+
+   .. code-block:: http
+
+      HTTP/1.1 302 Found
+      Location: http://sso.nootenboom.local?session_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+
+#. The authorization Server performs all the checks:
+
+   #. Retrieve the access token from the access token: `rfc7662 <https://tools.ietf.org/html/rfc7662>`_ (This endpoint has yet to be implemented)
+   #. Validate the identity from the session using the validate endpoint in Nuts Auth
+
+#. The authorization Server creates an internal URL and session and redirects the user
+#. The resource server shows the page within the client context from the access token
+
+
+Endpoints
+=========
+
+The an registry endpoint should use the following for the Nuts OAuth authorization server
+::
+
+  urn:ietf:rfc:3986:urn:oid:1.3.6.1.4.1.54851.1:nuts-oauth-authentication-server
+
+And the following type for the nuts SSO jump endpoint
+::
+
+  urn:ietf:rfc:3986:urn:oid:1.3.6.1.4.1.54851.1:nuts-sso-jump-endpoint
+
+TODO
```

Comment:
Maybe combine 2 API's into one? (login contract + jump token)

Reply:
The login contract consists of createContractSession and getContractStatus. This status might return the signed irma contract, but might not if the signing failed. Due to this splitted nature, I do not see an elegant way of combining the login contract with obtaining the jump token.